pkg/exporter/handler.go

```diff
@@ -1,28 +1,112 @@
 package exporter
 
 import (
+	"context"
+
 	v1 "github.com/re-cinq/cloud-carbon/pkg/types/v1"
 	bus "github.com/re-cinq/go-bus"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	api "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/metric"
 	"k8s.io/klog/v2"
 )
 
 type PrometheusEventHandler struct {
 	eventBus bus.Bus
+	meter    api.Meter
+	exporter prometheus.Exporter
 }
 
 func NewPrometheusEventHandler(eventBus bus.Bus) *PrometheusEventHandler {
+	exporter, err := prometheus.New()
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	meter := metric.NewMeterProvider(
+		metric.WithReader(exporter),
+	).Meter("cloud-carbon")
+
 	return &PrometheusEventHandler{
 		eventBus: eventBus,
+		meter:    meter,
 	}
 }
 
 func (c *PrometheusEventHandler) Apply(event bus.Event) {
-	// Make sure we got the right event
-	if _, ok := event.(*v1.EmissionsCalculated); ok {
-		// TODO: update the prometheus registry
 
+	e, ok := event.(v1.EmissionsCalculated)
+	if !ok {
+		klog.Errorf("PrometheusEventHandler got an unknown event: %T", event)
 		return
 	}
 
-	klog.Errorf("PrometheusEventHandler got an unknown event: %+v", event)
+	// setup emissions gauge
+	emissions, _ := c.meter.Float64ObservableGauge(
```

Comment:
If this is returning an error, we should handle it instead of ignoring it. But if it's some other datatype ignore me ( I didn't check the function)

Reply:
fair! I'll fix that :+1: